### PR TITLE
Adds ReceiptEmptyState Component and adds new component to MoneyRequestView

### PR DESCRIPTION
**Pull request by cdanwards**
### Fixed Issues

$  https://github.com/Expensify/App/issues/30663

This is the second PR for this issue. It adds the new `ReceiptEmptyState` component and also adds it to the `MoneyRequestView`.

### Tests

- [ ] Navigate to a Money Request that does _not_ contain a receipt attached
- [ ] Verify that the `ReceiptEmptyState` component is shown
- [ ] Within the `MoneyRequestView` set `hasErrors` to true
- [ ] Verify that the `ReceiptEmptyState` component has a red border
- [ ] Tap the `ReceiptEmptyState` component
- [ ] Upload an image via the Receipt Selector and after completion, verify that the `ReceiptEmptyState` component is no longer shown on the Money Request View

### Offline tests

- [ ] Navigate to a Money Request that does _not_ contain a receipt attached
- [ ] Verify that the `ReceiptEmptyState` component is shown
- [ ] Within the `MoneyRequestView` set `hasErrors` to true
- [ ] Verify that the `ReceiptEmptyState` component has a red border
- [ ] Tap the `ReceiptEmptyState` component
- [ ] Upload an image via the Receipt Selector and after completion, verify that the `ReceiptEmptyState` component is no longer shown on the Money Request View

### QA Steps

- [ ] Verify that no errors appear in the JS console
- [ ] Set the `Betas.canUseViolations` to true
- [ ] Navigate to a Money Request that does _not_ contain a receipt attached
- [ ] Verify that the `ReceiptEmptyState` component is shown
- [ ] Within the `MoneyRequestView` set `hasErrors` to true
- [ ] Verify that the `ReceiptEmptyState` component has a red border
- [ ] Tap the `ReceiptEmptyState` component
- [ ] Verify that the app navigates to the Receipt Selector
- [ ] Select an image
- [ ] Verify that the app navigates you to the Money Request view
- [ ] Verify that the correct image has been uploaded
- [ ] Verify that the `ReceiptEmptyState` component is no longer shown

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/8878532/37f7e055-828d-44ec-a3fd-d934ba45e74c

</details>

<details>
<summary>Android: mWeb Chrome</summary>
Screenshot 2023-11-10 at 4 05 14 PM

Screenshot 2023-11-10 at 4 05 28 PM

https://github.com/Expensify/App/assets/8878532/fbb2e54f-8bc9-421c-94b3-9cc532152397

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/8878532/72b788b8-afc3-4c31-a96e-e0b68aca239c

</details>

<details>
<summary>iOS: mWeb Safari</summary>
Screenshot 2023-11-10 at 4 09 31 PM

Screenshot 2023-11-10 at 4 09 51 PM

https://github.com/Expensify/App/assets/8878532/c48bcebb-cce6-40e6-80ce-fd5a2e630faf

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/8878532/d0b81268-8455-4fe0-b827-92c8d8520a38

</details>
<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/8878532/b9bbfe29-37ae-44bd-b1a0-88e22e06c1a7

</details>

### CODE REVIEW

**Review by pecanoro — CHANGES_REQUESTED** [review]

File: src/components/ReceiptEmptyState.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import React from 'react';
+import {View} from 'react-native';
+import styles from '@styles/styles';
+import variables from '@styles/variables';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import PressableWithoutFeedback from './Pressable/PressableWithoutFeedback';
+
+const propTypes = {
+    /** Whether or not there is a violation error */
```
Comment (cead22) [inline src/components/ReceiptEmptyState.js:11]: This component should know nothing about violations
```suggestion
    /** Whether or not there is an error */
```

File: src/components/ReceiptEmptyState.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    hasError: PropTypes.bool,
+
+    /** Callback to be called on onPress */
+    onPress: PropTypes.func,
+};
+
+const defaultProps = {
+    hasError: false,
+    onPress: undefined,
```
Comment (cead22) [inline src/components/ReceiptEmptyState.js:20]:
```suggestion
    onPress: () => {},
```

File: src/components/ReceiptEmptyState.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+// Create a component with the above instructions:
```
[the remark this answers is not in the document]
Comment (cead22) [reply src/components/ReceiptEmptyState.js:23]: Agreed, this should describe what the component is

**Review by cead22 — CHANGES_REQUESTED** [review]

File: src/components/ReceiptEmptyState.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (cdanwards) [reply src/components/ReceiptEmptyState.js:23]: Absolutely. This was a comment for myself I overlooked before committing. I'll update to correct JSDoc configuration

File: src/components/ReportActionItem/MoneyRequestView.js [hunk 4]
```diff
@@ -175,6 +176,14 @@ function MoneyRequestView({report, betas, parentReport, policyCategories, should
                         </View>
                     </OfflineWithFeedback>
                 )}
+                {!hasReceipt && Permissions.canUseViolations() && (
+                    <OfflineWithFeedback pendingAction={pendingAction}>
```
[the remark this answers is not in the document]
Comment (cdanwards) [reply src/components/ReportActionItem/MoneyRequestView.js]: After looking at `pendingActions` more I realized I misunderstood its purpose. Going to remove it!

**Review by pecanoro — DISMISSED** [review]

**Comment by pecanoro** [comment]
Let's see if puller bear gets us a c+ to review it

**Comment by pecanoro** [comment]
@cdanwards It seems we are missing screenshots in some platforms btw!

**Comment by pecanoro** [comment]
Ok, it seems puller bear is not going to work so I will ask for a C+ in the channel.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by cdanwards** [comment]
@parasharrajat it looks like the video you uploaded isn't loading correctly. Can you describe a little more what's going on for that BUG?

**Comment by cdanwards** [comment]
@pecanoro I'll upload those screenshots for mWeb.

**Comment by cdanwards** [comment]
@parasharrajat thank you! I'll take a look.

**Comment by cdanwards** [comment]
@parasharrajat I moved the styles to the pressable and removed that view so that the pressable didn't extend across the empty space. Let me know if there's anything else!

**Review by parasharrajat — CHANGES_REQUESTED** [review]
⬆️

**Comment by cdanwards** [comment]
>When testing the current behavior, it does not show the image placeholder if the request has no receipt at all so I am not sure if we want that or not.

@pecanoro following up on this comment. Is there more info you can provide around this? Are the betas set to allow you to use violations?

**Comment by cead22** [comment]
Good catch, in the issue I suggested the following

>if hasReceipt is false and permissions.canUseViolations() is true we show the ReceiptEmptyState component

But we should only show this receipt empty state if the money request can be edited, and it looks like we can use `canEdit` for that. How does that sound?

**Comment by cead22** [comment]
That works, but let's not block on this if this receipt empty state is only showing for the case in question for users on the beta.

I'll check with design to see if we should show a different image, or the same image with gray/secondary colors, or something else

**Review by parasharrajat — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.


:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by cead22 — APPROVED** [review]

**Review by pecanoro — APPROVED** [review]

**Comment by kavimuru** [comment]
@cdanwards @parasharrajat Could you help with the following  in the QA steps?

> Set the Betas.canUseViolations to true
> Within the MoneyRequestView set hasErrors to true

**Comment by cead22** [comment]
> Set the Betas.canUseViolations to true

@kavimuru for this, just use an email address @applause.expensifail.com since that domain is on all betas

> * Within the `MoneyRequestView` set `hasErrors` to true
> * [ ]  Verify that the `ReceiptEmptyState` component has a red border

You can skip these two steps actually

**Comment by cead22** [comment]
I approved but meant to test -- I shouldn't have approved, but testing isn't working for me. It looks like Permissions.canUseViolations isn't defined in src/components/ReportActionItem/MoneyRequestView.js, because we're not importing it. @cdanwards can you confirm and fix please, and add tests covering the case of a user on the beta, and a user not on the beta?

**Comment by cead22** [comment]
Let's also get Design's eyes on the border for this new component, as it looks a little bit too thick to me

File: src/components/ReportActionItem/MoneyRequestView.js [hunk 5]
```diff
@@ -176,6 +177,12 @@ function MoneyRequestView({report, betas, parentReport, policyCategories, should
                         </View>
                     </OfflineWithFeedback>
                 )}
+                {!hasReceipt && canEdit && !isSettled && Permissions.canUseViolations() && (
```
Comment (situchan) [inline src/components/ReportActionItem/MoneyRequestView.js:180]: This caused regressions:
- #31407
- #31418

`Permissions` is not imported. Anyone please fix these deploy blockers?

**Comment by MonilBhavsar** [comment]
> Looks like it was removed during the merge commit. why didn't it throw a lint warning?

Agree, let's create a new issue for this. It should throw in such cases

**Comment by cdanwards** [comment]
Thanks @parasharrajat. Sorry @cead22 for that oversight.

**Pull request merged**